src/cogs/jaegeraccounts.py

```diff
@@ -294,31 +299,52 @@ async def distribute_accounts(self, ctx: commands.Context, force="False"):
         for account in sheet_data.accounts:
-            if force == "force" or not account.is_booked:
+            if account.last_user in mentioned_users_accounts:
+                mentioned_users_accounts[account.last_user]["account"] = account
+            elif force == "force" or not account.is_booked:
                 available_accounts.append(account)
 
-        # TODO Need to figure out how to handle situations when one of mentioned users already has an account
-        if len(available_accounts) >= len(ctx.message.mentions):
-            accounts_to_assign = []
-            for member in ctx.message.mentions:
-                # Making sure account usage is evenly spread
-                random.shuffle(available_accounts)
+        accounts_to_assign = []      
+        for memberName, pairing in mentioned_users_accounts.items():
+            member = pairing["member"]
+            # If user already has previous account, check if its booked and send corresponding info
+            if "account" in pairing:
+                account = pairing["account"]
+                if account.is_booked:
+                    await member.send(f"You have already been assigned: `{account.name}`.\nPlease check your PMs for the login details.")
+                else:
+                    accounts_to_assign.append(account)
+                    member.send(embed=account.embed)
+                    await member.send(embed=account.embed)
+                    await ctx.author.send(f"Member {memberName} has been assigned account {account.name}.")
+            # If user doesnt have previously assigned account, find available and send
+            elif (len(available_accounts) > 0):
                 account = available_accounts.pop()
-                name = member.nick if member.nick is not None else member.name
-                account.last_user = name
+                account.last_user = memberName
                 accounts_to_assign.append(account)
                 await member.send(embed=account.embed)
-                await ctx.author.send(f"Member {name} has been assigned account {account.name}.")
-            await sheet_data.insert_bookings(self.bot, ctx, url, accounts_to_assign, 1)
-        else:
-            raise NoAccountsLeftException("There are not enought accounts in the google sheet to distribute this many accounts.")
+                await ctx.author.send(f"Member {memberName} has been assigned account {account.name}.")
+                # In case if ran out of accounts, notify requester about that
+                if len(available_accounts) == 0:
+                    await ctx.author.send("You have used all available account!")
+            else:
+                await ctx.author.send(f"Can not assign account to {memberName}, no more available accounts left!")
+
+        await sheet_data.insert_bookings(self.bot, ctx, url, accounts_to_assign, 1)
 
 def setup(bot):
-    bot.add_cog(AccountDistrubution(bot))
+    bot.add_cog(AccountDistrubution(bot))
```

Comment:
Idk why, but its best practice to leave a newline at the end of a file.
```suggestion
    bot.add_cog(AccountDistrubution(bot))

```